File: backend/app/models/recommendation.py
from sqlalchemy import Column, Integer, String, Float, DateTime, JSON, ForeignKey, Text, Boolean
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.core.database import Base


class Recommendation(Base):
    __tablename__ = "recommendations"
    
    id = Column(Integer, primary_key=True, index=True)
    location_id = Column(Integer, ForeignKey("locations.id"), nullable=False)
    
    # Recommendation details
    title = Column(String, nullable=False)
    description = Column(Text, nullable=False)
    category = Column(String)  # vegetation, infrastructure, policy, emergency
    priority = Column(String)  # high, medium, low
    
    # Impact metrics
    temperature_reduction = Column(Float)  # Expected reduction in Celsius
    cost_estimate = Column(Float)  # In INR
    implementation_time = Column(String)  # e.g., "6 months"
    benefit_cost_ratio = Column(Float)
    
    # AI confidence
    ai_confidence = Column(Float)
    shap_values = Column(JSON)  # Explainable AI values
    
    # Status
    status = Column(String, default="proposed")  # proposed, approved, implemented, completed
    implementation_date = Column(DateTime(timezone=True))
    
    extra_data = Column(JSON)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())


class HeatAlert(Base):
    __tablename__ = "heat_alerts"
    
    id = Column(Integer, primary_key=True, index=True)
    location_id = Column(Integer, ForeignKey("locations.id"), nullable=False)
    
    # Alert details
    alert_type = Column(String)  # heatwave, extreme_heat, advisory
    severity = Column(String)  # extreme, severe, moderate
    title = Column(String, nullable=False)
    message = Column(Text, nullable=False)
    
    # Time
    start_time = Column(DateTime(timezone=True), nullable=False)
    end_time = Column(DateTime(timezone=True), nullable=False)
    issued_at = Column(DateTime(timezone=True), server_default=func.now())
    
    # Conditions
    max_temperature = Column(Float)
    min_temperature = Column(Float)
    humidity = Column(Float)
    heat_index = Column(Float)
    
    # Affected area
    affected_population = Column(Integer)
    area_description = Column(String)
    
    # Actions
    recommended_actions = Column(JSON)
    is_active = Column(Boolean, default=True)
    
    extra_data = Column(JSON)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

File: mobile/lib/models/recommendation.dart
class Recommendation {
  final int id;
  final int locationId;
  final String title;
  final String description;
  final String category;
  final String priority;
  final double temperatureReduction;
  final double costEstimate;
  final String implementationTime;
  final double benefitCostRatio;
  final String status;

  Recommendation({
    required this.id,
    required this.locationId,
    required this.title,
    required this.description,
    required this.category,
    required this.priority,
    required this.temperatureReduction,
    required this.costEstimate,
    required this.implementationTime,
    required this.benefitCostRatio,
    required this.status,
  });

  factory Recommendation.fromJson(Map<String, dynamic> json) {
    return Recommendation(
      id: json['id'],
      locationId: json['location_id'],
      title: json['title'],
      description: json['description'],
      category: json['category'],
      priority: json['priority'],
      temperatureReduction: json['temperature_reduction'].toDouble(),
      costEstimate: json['cost_estimate'].toDouble(),
      implementationTime: json['implementation_time'],
      benefitCostRatio: json['benefit_cost_ratio'].toDouble(),
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'location_id': locationId,
      'title': title,
      'description': description,
      'category': category,
      'priority': priority,
      'temperature_reduction': temperatureReduction,
      'cost_estimate': costEstimate,
      'implementation_time': implementationTime,
      'benefit_cost_ratio': benefitCostRatio,
      'status': status,
    };
  }
}
